pkg/api/handlers/compat/images_build.go

```diff
@@ -573,14 +573,15 @@ func BuildImage(w http.ResponseWriter, r *http.Request) {
 		ulimits = m
 	}
 
+	pullPolicyMap := buildahDefine.PolicyMap
```

Comment:
maps are shared by reference, therefore you are modifying a global map from a different package. This could effect other code paths.
I suggest you do not modify the map at all and just add some extra if block to handle this case.